File: CreationalPatterns/AbstractFactory/abstract.js
var SportShirt = /** @class */ (function () {
    function SportShirt() {
    }
    SportShirt.prototype.hasButtons = function () {
        return false;
    };
    SportShirt.prototype.hasLongSleeves = function () {
        return false;
    };
    return SportShirt;
}());
var GalaShirt = /** @class */ (function () {
    function GalaShirt() {
    }
    GalaShirt.prototype.hasButtons = function () {
        return true;
    };
    GalaShirt.prototype.hasLongSleeves = function () {
        return true;
    };
    return GalaShirt;
}());
var SportShoe = /** @class */ (function () {
    function SportShoe() {
    }
    SportShoe.prototype.isElegantShoes = function () {
        return false;
    };
    SportShoe.prototype.isRunningShoes = function () {
        return true;
    };
    return SportShoe;
}());
var GalaShoe = /** @class */ (function () {
    function GalaShoe() {
    }
    GalaShoe.prototype.isElegantShoes = function () {
        return true;
    };
    GalaShoe.prototype.isRunningShoes = function () {
        return false;
    };
    return GalaShoe;
}());
var SportTrousers = /** @class */ (function () {
    function SportTrousers() {
    }
    SportTrousers.prototype.hasPockets = function () {
        return false;
    };
    SportTrousers.prototype.getClosureType = function () {
        return "Doesn't have closure type";
    };
    return SportTrousers;
}());
var GalaTrousers = /** @class */ (function () {
    function GalaTrousers() {
    }
    GalaTrousers.prototype.hasPockets = function () {
        return true;
    };
    GalaTrousers.prototype.getClosureType = function () {
        return "Closure Gala type";
    };
    return GalaTrousers;
}());
var SportClothesFactory = /** @class */ (function () {
    function SportClothesFactory() {
    }
    SportClothesFactory.prototype.createShoes = function () {
        return new SportShoe();
    };
    SportClothesFactory.prototype.createShirt = function () {
        return new SportShirt();
    };
    SportClothesFactory.prototype.createTrouser = function () {
        return new SportTrousers();
    };
    return SportClothesFactory;
}());
var GalaClothesFactory = /** @class */ (function () {
    function GalaClothesFactory() {
    }
    GalaClothesFactory.prototype.createShoes = function () {
        return new GalaShoe();
    };
    GalaClothesFactory.prototype.createShirt = function () {
        return new GalaShirt();
    };
    GalaClothesFactory.prototype.createTrouser = function () {
        return new GalaTrousers();
    };
    return GalaClothesFactory;
}());
// Creation and Logging
var sportFactory = new SportClothesFactory();
var galaFactory = new GalaClothesFactory();
var sportShirt = sportFactory.createShirt();
var sportShoe = sportFactory.createShoes();
var sportTrousers = sportFactory.createTrouser();
console.log("Sport Shirt:");
console.log("  Has Buttons:", sportShirt.hasButtons());
console.log("  Has Long Sleeves:", sportShirt.hasLongSleeves());
console.log("Sport Shoe:");
console.log("  Is Elegant Shoes:", sportShoe.isElegantShoes());
console.log("  Is Running Shoes:", sportShoe.isRunningShoes());
console.log("Sport Trousers:");
console.log("  Has Pockets:", sportTrousers.hasPockets());
console.log("  Closure Type:", sportTrousers.getClosureType());
console.log("\n---\n");
var galaShirt = galaFactory.createShirt();
var galaShoe = galaFactory.createShoes();
var galaTrousers = galaFactory.createTrouser();
console.log("Gala Shirt:");
console.log("  Has Buttons:", galaShirt.hasButtons());
console.log("  Has Long Sleeves:", galaShirt.hasLongSleeves());
console.log("Gala Shoe:");
console.log("  Is Elegant Shoes:", galaShoe.isElegantShoes());
console.log("  Is Running Shoes:", galaShoe.isRunningShoes());
console.log("Gala Trousers:");
console.log("  Has Pockets:", galaTrousers.hasPockets());
console.log("  Closure Type:", galaTrousers.getClosureType());

File: CreationalPatterns/AbstractFactory/abstract.cs
using System;

    public interface IShirt
    {
        bool HasLongSleeves();

        bool HasButtons();
    }

    public interface IShoe
    {
        bool IsElegantShoes();

        bool IsRunningShoes();
    }

    public interface ITrousers
    {
        bool HasPockets(); 
        string GetClosureType();
    }

    public interface IClothesFactory
    {
        IShirt CreateShirt();
        IShoe CreateShoes();
        ITrousers CreateTrousers();
    }

    public class SportShirt : IShirt
    {
        public bool HasLongSleeves()
        {
            return false;
        }

        public bool HasButtons()
        { 
            return false;        
        }
    }

    public class SportShoe : IShoe
    {
        public bool IsElegantShoes()
        {
            return false;
        }

        public bool IsRunningShoes()
        {
            return true;
        }
    }

    public class SportTrousers : ITrousers
    {
        public bool HasPockets()
        {
            return false;
        }

        public string GetClosureType()
        {
            return "Doesn't Have Closure Type";
        }
    }

    public class ElegantShoes : IShoe
    {
        public bool IsElegantShoes()
        {
            return true;
        }

        public bool IsRunningShoes()
        {
            return false;
        }
    }

    public class ElegantShirt : IShirt
    {
        public bool HasLongSleeves()
        {
            return true;
        }

        public bool HasButtons()
        {
            return true;
        }
    }

    public class ElegantTrousers : ITrousers
    {
        public bool HasPockets()
        {
            return true;
        }

        public string GetClosureType()
        {
            return "Closure Elegant Type";
        }
    }

    class SportClothesFactory : IClothesFactory
    {
        public IShirt CreateShirt()
        {
            return new SportShirt();

        }

        public IShoe CreateShoes()
        {
            return new SportShoe();
        }

        public ITrousers CreateTrousers()
        {
            return new SportTrousers();
        }
    }

    class ElegantClothesFactory : IClothesFactory
    {
        public IShirt CreateShirt()
        {
            return new ElegantShirt();
        }

        public IShoe CreateShoes()
        {
            return new ElegantShoes();
        }

        public ITrousers CreateTrousers()
        {
            return new ElegantTrousers();
        }
    }

     public class Program
    {
        static void Main(string[] args)
        {
                
             // Abstract Factory Method
            IClothesFactory SportsFactory = new SportClothesFactory();
            IClothesFactory ElegantFactory = new ElegantClothesFactory();

            IShirt sportShirt = SportsFactory.CreateShirt();
            IShoe sportShoes = SportsFactory.CreateShoes();
            ITrousers sportTrousers = SportsFactory.CreateTrousers();
            
            IShirt elegantShirt = ElegantFactory.CreateShirt();
            IShoe ElegantShoes = ElegantFactory.CreateShoes();
            ITrousers elegantTrousers = ElegantFactory.CreateTrousers();

            Console.WriteLine("Elegant clothes");
            Console.WriteLine("Has buttons " + elegantShirt.HasButtons());
            Console.WriteLine("Has Long sleeves " + elegantShirt.HasLongSleeves());
            Console.WriteLine("Is elegant shoes: " + ElegantShoes.IsElegantShoes());
            

        }
        
        
    }


